forage-android/src/main/java/com/joinforage/forage/android/ecom/services/vault/bt/BasisTheoryPinSubmitter.kt

```diff
@@ -61,7 +63,18 @@ internal class BasisTheoryPinSubmitter(
             bt.proxy.post(proxyRequest)
         }
 
-        return vaultToForageResponse(BTResponseParser(vaultResponse))
+        return try {
+            vaultToForageResponse(BTResponseParser(vaultResponse))
+        } catch (e: UnknownBTSuccessResponse) {
+            logger.e("Unknown success response from BasisTheory Vault.", e)
+            UnknownErrorApiResponse
+        } catch (e: UnknownForageFailureResponse) {
+            logger.e("Unknown error from Payments API.", e)
+            UnknownErrorApiResponse
+        } catch (e: Exception) {
+            logger.e("Request to Basis Theory failed.", e)
+            UnknownErrorApiResponse
+        }
```

Comment:
and this is where we catch the various exceptions that could arise from parsing BT responses